lib/elixir/lib/inspect.ex

```diff
@@ -380,21 +380,38 @@ defimpl Inspect, for: Map do
 end
 
 defimpl Inspect, for: Integer do
+  def inspect(term, %Inspect.Opts{base: :decimal, digit_separator: separator}) do
+    pretty_decimal(term, separator)
+  end
+
   def inspect(term, %Inspect.Opts{base: base}) do
     Integer.to_string(term, base_to_value(base))
     |> prepend_prefix(base)
   end
 
+  def pretty_decimal(n, separator) when n < 0 do
+    "-" <> pretty_decimal(abs(n), separator)
+  end
+
+  def pretty_decimal(n, separator) when n >= 1000 do
+    left = div(n, 1_000)
+    right = rem(n, 1_000)
```

Comment:
Haha, rub it in, why don't you ;) @lexmag: https://github.com/elixir-lang/elixir/pull/4795